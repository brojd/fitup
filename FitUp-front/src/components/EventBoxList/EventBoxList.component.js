import React, { PropTypes } from 'react';
import styles from './EventBoxList.stylesheet.css';
import Image from '../Image/Image.component';

class EventBoxList extends React.Component {

  cards_accepted() {
    return (this.props.currentGym.cards_accepted.map((card, i) => {
      if (card.accepted) {
        return (
          <span className='uk-margin-small-right uk-badge' key={i}>{card.name}</span>
        );
      }
    }));
  }
  
  prices() {
    return (this.props.currentGym.pricing.map((price, i) => {
      return (
        <div key={i}>{price.name}: {price.price} PLN</div>
      );
    }));
  }
  
  render() {
    return (
      <ul className='uk-list uk-list-space uk-list-line'>
        <li>TRAINER
          <span className="uk-float-right">
            <Image className={styles.classesInfoWrapper__trainerPhoto}
                   src={this.props.currentTrainer.photo_url}
                   alt='trainer_photo' width={50}/>
            {this.props.currentTrainer.name} {this.props.currentTrainer.surname}
          </span>
        </li>
        <li>GYM <span className="uk-float-right">{this.props.currentGym.name}</span></li>
        <li>
          ADDRESS
          <span className="uk-float-right">
            {`${this.props.currentGym.address.street} ${this.props.currentGym.address.number}, ` +
             `${this.props.currentGym.address.city}`}
          </span>
        </li>
        <li>CARDS ACCEPTED <span className="uk-float-right">{this.cards_accepted()}</span></li>
        <li>PRICES <span className="uk-float-right">{this.prices()}</span></li>
      </ul>
    );
  }
}

EventBoxList.propTypes = {
  currentTrainer: PropTypes.object,
  currentGym: PropTypes.object
};

EventBoxList.defaultProps = {
  currentTrainer: {},
  currentGym: {
    address: {
      city:''
    },
    cards_accepted: [''],
    pricing: [{}]
  }
};

export default EventBoxList;
