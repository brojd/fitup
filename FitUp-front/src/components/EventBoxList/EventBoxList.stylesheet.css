.classesInfoWrapper__trainerPhoto {
  border-radius: 4%;
  border: 1px solid white;
  margin-right: 6px; }

/*# sourceMappingURL=EventBoxList.stylesheet.css.map */
