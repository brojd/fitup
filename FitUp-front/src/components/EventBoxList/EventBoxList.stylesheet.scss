@import "../../../sass_variables/colors";

.classesInfoWrapper__trainerPhoto {
  border-radius: 4%;
  border: 1px solid white;
  margin-right: 6px;
}
