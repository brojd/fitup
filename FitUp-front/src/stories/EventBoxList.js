import React from 'react';
import { storiesOf} from '@kadira/storybook';
import EventBoxList from '../components/EventBoxList/EventBoxList.component';
import event from './mock_data/eventInstance';

storiesOf('EventBoxList', module)
  .add('event box list', () => (
    <div className='uk-width-2-10'>
      <EventBoxList currentTrainer={event.currentTrainer} currentGym={event.currentGym} />
    </div>
  ));

