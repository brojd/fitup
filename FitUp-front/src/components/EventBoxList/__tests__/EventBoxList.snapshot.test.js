import React from 'react';
import EventBoxList from '../EventBoxList.component';
import renderer from 'react-test-renderer';

test('EventBoxList renders correctly', () => {
  
  const CURRENT_TRAINER = {
    'name': 'Andrew',
    'surname': 'Kowalski',
    'about_me': 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation',
    'specialties': {
      'zumba': true,
      'yoga': true,
      'crossfit': false,
      'TBC': false,
      'pilates': false
    },
    'photo_url': 'http://trenujzmichalem.pl/wp-content/uploads/2014/01/ostr-slider-535x572.png',
    'gymId': '582ad891b402801294f4be25'
  };
  
  const CURRENT_GYM = {
    '_id': '582ad891b402801294f4be25',
    'email': 'gym1@gym1.pl',
    'password': 'gym1',
    'name': 'Fitness Academy',
    'address': {'city': 'Wrocław', 'postal_code': '50-001', 'street':'Długa', 'number':3},
    'description': 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.',
    'logo_url': 'https://pbs.twimg.com/profile_images/378800000513930765/1518318ade4947d639fca13c47b457f6.jpeg',
    'photo_urls': [
      'https://www.thegymgroup.com/fileadmin/uploads/gym/Photos/Gym_Leicester/Leicester14.jpg'
    ],
    'pricing': [
      {'name':'1 workout', 'price':15},
      {'name':'1 hour', 'price':40},
      {'name':'1 month', 'price':130}
    ],
    'cards_accepted': [
      {'name':'Multisport', 'accepted':true}, {'name':'ActiveSport', 'accepted':true}, {'name':'OKsystem', 'accepted':true}, {'name':'FitWro', 'accepted':false}, {'name':'FitProfit', 'accepted':false}
    ]
  };
  
  const component = renderer.create(
    <EventBoxList currentTrainer={CURRENT_TRAINER}
                  currentGym={CURRENT_GYM} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
