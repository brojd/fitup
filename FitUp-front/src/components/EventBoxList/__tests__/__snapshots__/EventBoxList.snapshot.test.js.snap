exports[`test EventBoxList renders correctly 1`] = `
<ul
  className="uk-list uk-list-space uk-list-line">
  <li>
    TRAINER
    <span
      className="uk-float-right">
      <img
        alt="trainer_photo"
        className="classesInfoWrapper__trainerPhoto"
        height={150}
        onClick={undefined}
        src="http://trenujzmichalem.pl/wp-content/uploads/2014/01/ostr-slider-535x572.png"
        width={50} />
      Andrew
       
      Kowalski
    </span>
  </li>
  <li>
    GYM 
    <span
      className="uk-float-right">
      Fitness Academy
    </span>
  </li>
  <li>
    ADDRESS
    <span
      className="uk-float-right">
      Długa 3, Wrocław
    </span>
  </li>
  <li>
    CARDS ACCEPTED 
    <span
      className="uk-float-right">
      <span
        className="uk-margin-small-right uk-badge">
        Multisport
      </span>
      <span
        className="uk-margin-small-right uk-badge">
        ActiveSport
      </span>
      <span
        className="uk-margin-small-right uk-badge">
        OKsystem
      </span>
    </span>
  </li>
  <li>
    PRICES 
    <span
      className="uk-float-right">
      <div>
        1 workout
        : 
        15
         PLN
      </div>
      <div>
        1 hour
        : 
        40
         PLN
      </div>
      <div>
        1 month
        : 
        130
         PLN
      </div>
    </span>
  </li>
</ul>
`;
